producer/writer/message.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
Should this file be called `metadata.go` and you can move `metadataFromProto` here?